src/test/ValueClassesTests.scala

```diff
@@ -2,13 +2,63 @@ package magnolia1.tests
 
 import magnolia1.*
 import magnolia1.examples.*
+import scala.annotation.StaticAnnotation
 
-/** TODO: Support for value classes is missing for scala3 branch. Eventually
-  * refactor and uncomment the tests below once the feature is implemented.
+/** Supports mirrorless value classes derivation for non-generic products with annotations.
+ * TODO:
+    1) non-product derivation
+    2) generic derivation
+    3) access modifiers
   */
 class ValueClassesTests extends munit.FunSuite:
   import ValueClassesTests.*
 
+  test("Print foo") {
+    given Print[Int] = _.toString
+    val res = Print.derived[VC].print(VC(555))
+    assertEquals(res, "VC(555)")
+  }
+
+  test("Print bar") {
+    given Print[Int] = _.toString
+    val res = Print.derived[Bar].print(Bar(666))
+    assertEquals(res, "Bar(666)")
+  }
+
+  test("semi default for VC") {
+    given SemiDefault[VC] = SemiDefault.derived
+    val res = summon[SemiDefault[VC]].default
+    assertEquals(res, VC(0))
+  }
+
+  test("SemiDefault for BigBox value class") {
+    given SemiDefault[BigBox] = SemiDefault.derived
+    val res = summon[SemiDefault[BigBox]].default 
+    assertEquals(res, BigBox(NormalBox(SmallBox(TinyBox(0)))))
+  }
+
+  test("Semi default for Foo") {
+    given SemiDefault[Foo] = SemiDefault.derived
+    val res = summon[SemiDefault[Foo]].default
+    assertEquals(res, Foo(0))
+  }
+
+  test("show for Foo") {
+    val res = Show.derived[Foo].show(Foo(33))
+    assertEquals(res, "Foo{MyAnnotation(0)}(k{MyAnnotation(1)}{MyAnnotation(2)}=33)")
+  }
+
+  test("Print default") {
+    given Print[Int] = _.toString
+    val res = Print.derived[WithDefault].print(WithDefault())
+    assertEquals(res, "WithDefault(123)")
+  }
+
+  test("Semi default") {
+    val res = SemiDefault.derived[WithDefault].default
+    assertEquals(res, WithDefault(123))
+  }
+
   // test("serialize a value class") {
```

Comment:
what are we missing to enable these tests?

Reply:
ah ok I see above :)

Reply:
maybe add a comment fo reach test why it fails?

Reply:
Added.

Reply:
Thanks - but what's a product value class? Do you mean case classes? If so, `ServiceName1` is a case class, but it is disabled?